sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp

```diff
@@ -162,7 +163,7 @@ namespace Azure { namespace Messaging { namespace EventHubs {
     }
 
     Azure::Core::Amqp::Models::AmqpMessage CreateBatchEnvelope(
-        std::shared_ptr<Azure::Core::Amqp::Models::AmqpMessage const> const& message)
+        std::shared_ptr<Azure::Core::Amqp::Models::AmqpMessage const> const& message) const
```

Comment:
haw many more const can we add ? const ref to a const pointer to a const object :)

Reply:
I'm 100% sure we can add a few more. Let me see what I can do.

FWIW, the new `const` which was added was actually a suggestion from static code analysis which suggested I add `const` to the method because it doesn't modify any class members.

The others are because we don't want the caller to pass in a copy of the message (that's the final const&). The other is that the message contained in the pointer should be treated as a constant because this method is not allowed to modify the AmqpMethod within the shared_ptr.